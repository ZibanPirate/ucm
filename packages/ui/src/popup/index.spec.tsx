import { fireEvent, render, screen } from "@testing-library/react";
import { FC, useState } from "react";

import { Popup } from ".";

describe(`Testing component '${Popup.name}' :`, () => {
  it(`should render it shown, then hid it`, () => {
    const PageComponent: FC = () => {
      const [shown, setShown] = useState(true);
      return (
        <Popup
          data-testid="popup"
          containerProps={{ "data-testid": "popup-container" }}
          shown={shown}
          onClose={() => setShown(false)}
        />
      );
    };

    const { container } = render(<PageComponent />);
    expect(container).toMatchSnapshot("Shown");

    const popup = screen.getByTestId("popup");
    fireEvent.click(popup);
    expect(container).toMatchSnapshot("Stays shown");

    const popupContainer = screen.getByTestId("popup-container");
    fireEvent.click(popupContainer);
    expect(container).toMatchSnapshot("Hidden");
  });

  it(`should render it with no optional props`, () => {
    const { container } = render(<Popup data-testid="popup" shown={true} onClose={() => null} />);
    const popup = screen.getByTestId("popup");
    fireEvent.click(popup);
    expect(container).toMatchSnapshot();
  });
});
