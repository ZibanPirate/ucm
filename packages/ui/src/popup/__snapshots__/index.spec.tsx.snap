// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing component 'Popup' : should render it shown, then hid it: Hidden 1`] = `
<div>
  <div
    data-testid="popup-container"
    style="display: none; position: fixed; top: 0px; left: 0px; width: 100vw; height: 100vh; background-color: rgba(255, 255, 255, 0.933); overflow: auto; text-align: center;"
  >
    <div
      data-testid="popup"
      style="text-align: start; display: inline-block; background-color: lightgray; margin: 3rem;"
    />
  </div>
</div>
`;

exports[`Testing component 'Popup' : should render it shown, then hid it: Shown 1`] = `
<div>
  <div
    data-testid="popup-container"
    style="display: initial; position: fixed; top: 0px; left: 0px; width: 100vw; height: 100vh; background-color: rgba(255, 255, 255, 0.933); overflow: auto; text-align: center;"
  >
    <div
      data-testid="popup"
      style="text-align: start; display: inline-block; background-color: lightgray; margin: 3rem;"
    />
  </div>
</div>
`;

exports[`Testing component 'Popup' : should render it shown, then hid it: Stays shown 1`] = `
<div>
  <div
    data-testid="popup-container"
    style="display: initial; position: fixed; top: 0px; left: 0px; width: 100vw; height: 100vh; background-color: rgba(255, 255, 255, 0.933); overflow: auto; text-align: center;"
  >
    <div
      data-testid="popup"
      style="text-align: start; display: inline-block; background-color: lightgray; margin: 3rem;"
    />
  </div>
</div>
`;

exports[`Testing component 'Popup' : should render it with no optional props 1`] = `
<div>
  <div
    style="display: initial; position: fixed; top: 0px; left: 0px; width: 100vw; height: 100vh; background-color: rgba(255, 255, 255, 0.933); overflow: auto; text-align: center;"
  >
    <div
      data-testid="popup"
      style="text-align: start; display: inline-block; background-color: lightgray; margin: 3rem;"
    />
  </div>
</div>
`;
